src/coordination/coordinator_state_manager.cpp

```diff
@@ -80,25 +80,32 @@ void to_json(nlohmann::json &j, cluster_config const &cluster_config) {
                      {kUserCtx, cluster_config.get_user_ctx()}};
 }
 
+auto CoordinatorStateManager::HandleVersionMigration() -> void {
+  auto const version = memgraph::coordination::GetOrSetDefaultVersion(
+      durability_, kStateManagerDurabilityVersionKey, static_cast<int>(kActiveStateManagerDurabilityVersion), logger_);
+
+  if (version == static_cast<int>(StateManagerDurabilityVersion::kV1)) {
+    throw VersionMigrationException(
+        "Version migration for state manager from V1 to V2 is not supported. Cleanup all high availability directories "
```

Comment:
Maybe we can solve this in next PR, this one is already big, but I thought about it and as I explained I don't see how because:
- we can't make nuRAFT update all server configs for all instances at once
- we would need to update this manually, which means coordinators would need to talk and this would be step 0 of initialization and migration -> this is possible but let's put it off to the next PR and I can make it work